pkg/kubectl/apply/strategy/retain_keys_visitor.go

```diff
@@ -16,4 +16,62 @@ limitations under the License.
 
 package strategy
 
-// TODO: Write this
+import (
```

Comment:
This is really wonderful. A new strategy in 77 lines, copyright and imports included.